cleangoldenami/main.go

```diff
@@ -115,26 +183,69 @@ func getPublicImages(ec2Client DescribeImagesClient) ([]ec2Types.Image, error) {
 	return describeImagesResponse.Images, nil
 }
 
-func getOldestImage(images []ec2Types.Image) (ec2Types.Image, error) {
+func hasMatchingTag(image ec2Types.Image, tag string) bool {
+	for _, t := range image.Tags {
+		if *t.Key == "version" && *t.Value == tag {
+			return true
+		}
+	}
+	return false
+}
+
+func hasMatchingArchitecture(image ec2Types.Image, architecture string) bool {
+	return string(image.Architecture) == architecture
+}
+
+// filterImages filters AWS AMIs based on their version tag and architecture.
+func filterImages(images []ec2Types.Image, architecture string, tag string) []ec2Types.Image {
+	var filteredImages []ec2Types.Image
+	for _, image := range images {
+
+		if hasMatchingTag(image, tag) && hasMatchingArchitecture(image, architecture) {
+			filteredImages = append(filteredImages, image)
+		}
+	}
+	return filteredImages
+}
+
+// getOldestImage retrieves the oldest AWS AMI based on creation date
+func getOldestImage(images []ec2Types.Image) (int, ec2Types.Image, error) {
 	var oldestImage ec2Types.Image
 	var oldestImageTimestamp int64
-	for _, image := range images {
+	var oldestImageIndex = -1
+
+	if len(images) <= 1 {
+		return -1, oldestImage, nil
+	}
+	for i, image := range images {
 		creationTime, err := time.Parse(timeLayout, *image.CreationDate)
 		if err != nil {
-			return ec2Types.Image{}, fmt.Errorf("error parsing timestamp %v for image %v: %w", *image.CreationDate, image.ImageId, err)
+			return -1, ec2Types.Image{}, fmt.Errorf("error parsing timestamp %v for image %v: %w", *image.CreationDate, image.ImageId, err)
 		}
 		if creationTimeUnix := creationTime.Unix(); oldestImage.ImageId == nil || creationTimeUnix < oldestImageTimestamp {
 			oldestImage = image
 			oldestImageTimestamp = creationTimeUnix
+			oldestImageIndex = i
 		}
 	}
-	return oldestImage, nil
+	if oldestImageIndex >= 0 {
+		return oldestImageIndex, oldestImage, nil
+
+	}
+
+	return -1, oldestImage, nil
```

Comment:
If I'm reading correctly, I believe this would be functionally equivalent and save a few lines.
```suggestion
	return oldestImageIndex, oldestImage, nil
```